src/app/contexts/layout.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import buildContext from '~/components/jsx-helpers/build-context';
+import loadable from 'react-loadable';
+import LoadingPlaceholder from '~/components/loading-placeholder/loading-placeholder';
+
+// Webpack wasn't able to make the dynamic strings work in the Context value.
+type LayoutName = 'default' | 'landing';
+const loaders = {
+    default: () => import('~/layouts/default/default'),
+    landing: () => import('~/layouts/landing/landing')
+};
+
+type LayoutParameters = {
+    name: LayoutName;
+    data?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+}
+const defaultLayoutParameters: LayoutParameters = {name: 'default'};
+
+function useContextValue() {
+    const [layoutParameters, setLayoutParameters] = React.useReducer(
+        (state: LayoutParameters, newState: LayoutParameters) => {
+            if (newState === undefined) {
+                return defaultLayoutParameters;
+            }
+            if (state.name === newState.name && (
+                JSON.stringify(state.data) === JSON.stringify(newState.data) ||
+                state.name === 'default'
+            )) {
+                return state;
+            }
+            return newState;
```

Comment:
why do we have special handling for the 'default' layout here? shouldn't it be:
```suggestion
            if (state.name === newState.name && JSON.stringify(state.data) === JSON.stringify(newState.data)) {
                return state;
            }
            return newState;
```

also JSON.stringify will give you inconsistent results sometimes, [ts-utils has a utility](https://github.com/openstax/typescript-service-template/blob/main/packages/utils/src/misc/hashValue.ts) that handles hashing objects consistently, [deep-equal](https://www.npmjs.com/package/deep-equal) is also a very small dependency you could use here (that we are already installing anyway)